packages/rolldown/src/builtin-plugin/utils.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  BindingBuiltinPlugin,
+  BindingCallableBuiltinPlugin,
+  isCallableCompatibleBuiltinPlugin as isCallableCompatibleBuiltinPluginInternal,
+} from '../binding'
+
+import { BuiltinPlugin } from './constructors'
+
+type BindingCallableBuiltinPluginLike = {
+  [K in keyof BindingCallableBuiltinPlugin]: BindingCallableBuiltinPlugin[K]
+}
+
+export function makeBuiltinPluginCallable(plugin: BuiltinPlugin) {
```

Comment:
@sapphi-red Is this a hack could replaced with normal way?

All builtin-plugins are now exposed as a function I think.

I notice it's used in here: https://github.com/rolldown/vite/blob/8a8a50635e526e89ecb19882fe5f7608a0a02ee3/packages/vite/src/node/server/environment.ts#L175

Reply:
I guess no?
You cannot do:
```js
const plugin = viteResolvePlugin()
plugin.resolveId(id, importer) // this
```
without it. So I'm doing:
```js
const pluginOriginal = viteResolvePlugin()
const plugin = makeBuiltinPluginCallable(pluginOriginal)
plugin.resolveId(id, importer)
```

Reply:
Since we return a `class ViteResolvePlugin`, does manually add an `resolveId` method in `class ViteResolvePlugin` works?